docs/development/pigs/pig-009.rst

```diff
@@ -0,0 +1,238 @@
+accuracy. The need for this is unclear.
+
+
+MapEventSampler
+===============
+
+To support sampling from the predicted number of counts maps for source as well as background maps, we propose to
+introduce a `MapEventSampler` class. The responsibility of this class is to handle the physical coordinate transformations,
+time dependence of the model, output data structures and sampling of the total number of observed events.
+
+Starting from a predicted number of counts map (e.g. computed by the `MapEvaluator`):
+
+.. code::
+
+    src = SkyModel()
+    evaluator = MapEvaluator(src, exposure, psf=None, edisp=None)
+    npred_map = evaluator.compute_npred()
+
+The `MapEventSampler` takes this map and a lightcurve model as an input and allows to sample from it with the `.sample()`
+method:
+
+.. code::
+
+    lightcurve = LightCurveTableModel() # or PhaseCurveTableModel()
+    src_sampler = MapEventSampler(npred_map, lightcurve, t_min=, t_max=)
+
+    # draw total number of events
+    n_events = src_sampler.npred_total()
+
+    # draw events as an astropy table
+    events_src = map_sampler.sample()
+
+If no lightcurve is provided a constant rate can be assumed e.g. for background models:
+
+.. code::
+
+    bkg_map = BackgroundModel().map
+    bkg_sampler = MapEventSampler(bkg_map, t_min=, t_max=)
+    events_bkg = bkg_sample.sample()
+
+
+It remains to answer the question what to do when the source lightcurve is shorter than the total exposure time: do we
+repeat the temporal information or do we just consider the remaining time as a flat lightcurve?
+
+
+IRFMapSampler
+=============
+
+To handle correctly the spatially varying IRFs, we propose to introduce an `IRFMapSampler` class. Based on the `PSFMap`
+and `EDispMap` as an input it interpolates the "correct" IRF at the position of a given event and applies it. In more detail
+the class will calculate the psf and the energy dispersion for each event true position and true energy. The reconstructed
+energies and positions are then appended to the event list. The IRFs are assumed to be constant and not time-dependent.
+The time dependency will be handled higher level, by a loop over good time-intervals. The class would be used as following:
+
+.. code::
+
+    events = vstack([events_src_1, events_src_2, ..., events_src_N])
+    irf_sampler = IRFMapSampler(psf_map, edisp_map)
+
+    events = irf_sampler.sample_psf(events)
+    events = irf_sampler.sample_edisp(events)
+
+Alternatively the IRFs could be applied per model component and be handled in the `MapEventSampler`. However in this case
+the IRFs would not be applied correctly for e.g. very extended sources, because the IRF changes over region of the source.
+If we find the approach with the `IRFMapSampler` to be not performant enough or problematic in memory use, it can be still
+used as a simple fallback solution.
+
+MapDatasetEventSampler?
```

Comment:
The "?" was intended only for us, because we were not sure yet if this class would has been included or not in the PIG. Since we decided to keep it, we now removed the question mark from the class name.